Source/DafnyCore/Rewriters/PrecedenceLinter.cs

```diff
@@ -156,15 +166,11 @@ class PrecedenceLinterVisitor : TopDownVisitor<LeftMargin> {
         VisitRhsComponent(expr.tok, letExpr.Body, "body of let-expression");
         return false; // indicate that we've already processed expr's subexpressions
 
-      } else if (expr is OldExpr or FreshExpr or UnchangedExpr or DatatypeValue or DisplayExpression or MapDisplayExpr) {
+      } else if (expr is OldExpr or FreshExpr or UnchangedExpr or DisplayExpression or MapDisplayExpr) {
         // In these expressions, all subexpressions are contained in parentheses, so there's no risk of precedence confusion
         expr.SubExpressions.ForEach(VisitIndependentComponent);
         return false; // indicate that we've already processed expr's subexpressions
 
-      } else if (expr is FunctionCallExpr functionCallExpr) {
-        return VisitComponentsAsIndependentExceptOne(expr, functionCallExpr.Receiver, st);
-      } else if (expr is ApplyExpr applyExpr) {
-        return VisitComponentsAsIndependentExceptOne(expr, applyExpr.Function, st);
```

Comment:
If we no longer expect `DatatypeValue` and `FunctionCallExpr` to ever arise at this time, then:
* We could still look for them and do a `Contract.Assert(false);` (or something similar) if they ever occur.
* Is `VisitComponentAsIndependentExceptOne` now unused?